data.js

```diff
@@ -198,6 +198,12 @@ function save (callback) {
 
         // for update queries
 
+        if (err) {
+            callback("Update error: " + err);
+            self.emit('saved', err);
```

Comment:
I worry a little that this will crash things that are listening to `saved` event. Maybe renaming it to `saved_error` would be better...

However, I do not remember a use case...